src/content/posts/docker_volume_introduction.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "Use Docker's volumes to build a persistent log system"
+summary: "Persist data after container's death or share data accross containers through Docker's volume feature"
+date: 2023-09-19T12:00:00+0200
+lastUpdate: 2024-04-12T20:00:00+0200
+tags: ["docker", "volume", "log"]
+author: FireFrozen
+draft: false
+---
+
+Before reading this article, it's recommended to have a minimum knowledge of `docker`.  
+See, for example, the previous blog on docker: [Get Started with Docker](https://iscsc.fr/posts/short-docker-introduction/) by `ctmbl`.
+
+> **Disclaimer**:  
+> Note that in Docker, **two ways exist to persist/share data** with the host: **volumes** and **bind mounts**.  
+> While some key differences exist between both, they are not relevant in most of our basic use cases.  
+> We'll then **not distinguish one from the other**, some of what we say might better applied to bind mounts but **we'll only speak of volumes**.  
+> For more information see [Manage data in Docker](https://docs.docker.com/storage/).
+
+## What is a Docker's volume?
+
+A docker's volume is the action of **mounting an existing file or directory** of the host (the computer running the container) **into a container**. To rephrase, it enables a container to **share a memory space with the host**. So both the container and the user can access and modify it in real time and because it's not "contained" in the container but only mounted, it'll **persist after the container's death**.
```

Comment:
> So both the container and the user can access and modify it in real time

Mmh that is right but they should not
According to [the link you put just above](https://docs.docker.com/storage/#choose-the-right-type-of-mount):

> - Volumes are stored in a part of the host filesystem which is managed by Docker (/var/lib/docker/volumes/ on Linux). Non-Docker processes should not modify this part of the filesystem. Volumes are the best way to persist data in Docker.
> - Bind mounts may be stored anywhere on the host system. They may even be important system files or directories. Non-Docker processes on the Docker host or a Docker container can modify them at any time.

You explained pretty well in the two sentences before so I would just remove this and rephrase like this:
> Because files inside the volumes are stored on the host, they will persist after the container's death.

Reply:
When I say `both the container and the user can access and modify it in real time` I mean the **corresponding file in the host** because we're presenting `-v <path/to/host>:<path/to/cont>` syntax (which isn't a volume but a bind mount strictly speaking), [**of course never touch anything in `/var/lib/docker`**](https://forums.docker.com/t/write-from-host-to-volume/47274)